pkg/image/api/validation/validation.go

```diff
@@ -24,17 +56,31 @@ func ValidateImage(image *api.Image) errors.ValidationErrorList {
 func ValidateImageRepositoryMapping(mapping *api.ImageRepositoryMapping) errors.ValidationErrorList {
 	result := errors.ValidationErrorList{}
 
-	if len(mapping.DockerImageRepository) == 0 {
-		result = append(result, errors.NewFieldRequired("DockerImageRepository", mapping.DockerImageRepository))
+	hasRepository := len(mapping.DockerImageRepository) != 0
+	hasName := len(mapping.Name) != 0
+	switch {
+	case hasRepository:
+		_, _, _, _, err := api.SplitDockerPullSpec(mapping.DockerImageRepository)
+		if err != nil {
+			result = append(result, errors.NewFieldInvalid("dockerImageRepository", mapping.DockerImageRepository, err.Error()))
+		}
+	case hasName:
```

Comment:
Yes.  You can set name, or docker image repository, and name takes priority.  That's for backwards compat with existing docker-registries.  In v1beta2 we'll change that behavior prolly.


> On Jan 19, 2015, at 12:14 PM, Andy Goldstein <notifications@github.com> wrote:
> 
> In pkg/image/api/validation/validation.go:
> 
> > @@ -24,17 +56,31 @@ func ValidateImage(image *api.Image) errors.ValidationErrorList {
> >  func ValidateImageRepositoryMapping(mapping *api.ImageRepositoryMapping) errors.ValidationErrorList {
> >  	result := errors.ValidationErrorList{}
> >  
> > -	if len(mapping.DockerImageRepository) == 0 {
> > -		result = append(result, errors.NewFieldRequired("DockerImageRepository", mapping.DockerImageRepository))
> > +	hasRepository := len(mapping.DockerImageRepository) != 0
> > +	hasName := len(mapping.Name) != 0
> > +	switch {
> > +	case hasRepository:
> > +		_, _, _, _, err := api.SplitDockerPullSpec(mapping.DockerImageRepository)
> > +		if err != nil {
> > +			result = append(result, errors.NewFieldInvalid("dockerImageRepository", mapping.DockerImageRepository, err.Error()))
> > +		}
> > +	case hasName:
> If the mapping has a DockerImageRepository but no Name, you won't get a validation error. Was that intentional?
> 
> —
> Reply to this email directly or view it on GitHub.
>